lib/streamy/errors/publication_failed_error.rb

```diff
@@ -1,10 +1,11 @@
 module Streamy
   class PublicationFailedError < StandardError
-    attr_reader :event_params
+    attr_reader :event_params, :error_class
 
     def initialize(original_error, event)
       @event_params = event.to_params
-      super "Failed publishing event: #{event_params} \n#{original_error.class} - #{original_error.message}"
+      @error_class = original_error.class
+      super "Failed publishing event: #{event_params} \n#{error_class} - #{original_error.message}"
```

Comment:
If you just need it for the error message, maybe we can omit the instance var + reader and go:

```suggestion
      super "Failed publishing event: #{event_params} \n#{original_error.class} - #{original_error.message}"
```

If you need to be able to access the original error programmatically from the outside, maybe you can do `publication_failed_error.cause.class` instead of adding a reader?

Reply:
Ok good point... I don't think I need this at all ... I forgot about `cause`